qa/rpc-tests/test.py

```diff
@@ -0,0 +1,235 @@
+    'wallet_shieldcoinbase_sapling.py',
+    'wallet_listreceived.py',
+    'wallet.py',
+    'wallet_overwintertx.py',
+    'wallet_persistence.py', 
+    'wallet_nullifiers.py',
+    'wallet_1941.py',
+    'wallet_addresses.py',
+    'wallet_sapling.py',
+    'wallet_listnotes.py',
+    'mergetoaddress_sprout.py',
+    'mergetoaddress_sapling.py',
+    'listtransactions.py',
+    'mempool_resurrect_test.py',
+    'txn_doublespend.py',
+    'txn_doublespend.py --mineblock',
+    'getchaintips.py',
+    'rawtransactions.py',
+    'rest.py',
+    'mempool_spendcoinbase.py',
+    'mempool_reorg.py',
+    'mempool_tx_input_limit.py',
+    'mempool_nu_activation.py',
+    'mempool_tx_expiry.py',
+    'httpbasics.py',
+    'zapwallettxes.py',
+    'proxy_test.py',
+    'merkle_blocks.py',
+    'fundrawtransaction.py',
+    'signrawtransactions.py',
+    'signrawtransaction_offline.py',
+    'walletbackup.py',
+    'key_import_export.py',
+    'nodehandling.py',
+    'reindex.py',
+    'decodescript.py',
+    'blockchain.py',
+    'disablewallet.py',
+    'zcjoinsplit.py',
+    'zcjoinsplitdoublespend.py',
+    'zkey_import_export.py',
+    'reorg_limit.py',
+    'getblocktemplate.py',
+    'bip65-cltv-p2p.py',
+    'bipdersig-p2p.py',
+    'p2p_nu_peer_management.py',
+    'rewind_index.py',
+    'p2p_txexpiry_dos.py',
+    'p2p_txexpiringsoon.py',
+    'p2p_node_bloom.py',
+    'regtest_signrawtransaction.py',
+    'finalsaplingroot.py'
+)
+
+EXTENDED_SCRIPTS=(
+    'getblocktemplate_longpoll.py',
+    'getblocktemplate_proposals.py',
+    'pruning.py',
+    'forknotify.py',
+    'hardforkdetection.py',
+    'invalidateblock.py',
+    'keypool.py',
+    'receivedby.py',
+    'rpcbind_test.py',
+#   'script_test.py',
+    'smartfees.py',
+    'maxblocksinflight.py',
+    'invalidblockrequest.py',
+#    'forknotify.py',
+    'p2p-acceptblock.py'
+)
+
+NON_SCRIPTS =(
+    'rpc-tester.py'
+)
+
+def cleanup_failed_tests():
+    pass
+
+def run_test_scripts(test_list, args):
+    fail_list = []
+    pass_list = []
+    stats = []
+
+    total_tests = len(test_list)
+    num_tests_ran = 0
+
+    logging.info("=== Starting RPC Tester  ===")
+    total_stime = time.time()
+    for file in test_list:
+        logging.info("--- Running test : %s ---", file)
+        stats.append(file)
+        #might want to add routine to properly clean SIGTERM tests so they don't conga line the rest
+        try:
+            start_time = time.time()
+            p1 = Popen(['python', file])
+            buffer = p1.communicate()[0]
+        except:
+            fail_list.append(file)
+        
+        elapsed_time = time.time() - start_time
+        logging.info("Elapsed time: %s", time.strftime("%H:%M:%S", time.gmtime(elapsed_time)))
+        stats.append(time.strftime("%H:%M:%S", time.gmtime(elapsed_time)))
+       
+        if p1.returncode != 0:
+            fail_list.append(stats)
+        else:
+            pass_list.append(stats)
```

Comment:
I don't understand this -- `stats` is always the same object within this function, so we're appending that object multiple times?